functests/1_performance/netqueues.go

```diff
@@ -0,0 +1,359 @@
+							},
+						},
+					}
+					profiles.UpdateWithRetry(profile)
+					Eventually(func() bool {
+						err := checkDeviceSupport(workerRTNodes, devices)
+						Expect(err).ToNot(HaveOccurred())
+						return true
+					}, cluster.ComputeTestTimeout(240*time.Second, RunningOnSingleNode), testPollInterval*time.Second).Should(BeTrue())
+				}
+				for _, size := range devices {
+					if size == getReservedCPUSize(profile.Spec.CPU) {
+						count++
+					}
+				}
+				if count == 0 {
+					Skip("Skipping Test: Unable to set Network queue size to reserved cpu count")
+				}
+				//Verify the tuned profile is created on the worker-cnf nodes:
+				tunedCmd := []string{"bash", "-c",
+					fmt.Sprintf("cat /etc/tuned/openshift-node-performance-performance/tuned.conf | grep devices_udev_regex")}
+				for _, node := range workerRTNodes {
+					tunedPod := getTunedPod(&node)
+					out, err := pods.ExecCommandOnPod(tunedPod, tunedCmd)
+					deviceExists := strings.ContainsAny(string(out), device)
+					Expect(deviceExists).To(BeTrue())
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+	})
+})
+
+//Check if the device support multiple queues
+func checkDeviceSupport(workernodes []corev1.Node, devices map[string]int) error {
+	cmdGetPhysicalDevices := []string{"find", "/sys/class/net", "-type", "l", "-not", "-lname", "*virtual*", "-printf", "%f "}
+	var channelCurrentCombined int
+	var err error
+	for _, node := range workernodes {
+		tunedPod := getTunedPod(&node)
+		phyDevs, err := pods.ExecCommandOnPod(tunedPod, cmdGetPhysicalDevices)
+		Expect(err).ToNot(HaveOccurred())
+		for _, d := range strings.Split(string(phyDevs), " ") {
+			if d == "" {
+				continue
+			}
+			_, err := pods.ExecCommandOnPod(tunedPod, []string{"ethtool", "-l", d})
+			if err == nil {
+				cmdCombinedChannelsCurrent := []string{"bash", "-c",
+					fmt.Sprintf("ethtool -l %s | sed -n '/Current hardware settings:/,/Combined:/{s/^Combined:\\s*//p}'", d)}
+				out, err := pods.ExecCommandOnPod(tunedPod, cmdCombinedChannelsCurrent)
+				if strings.Contains(string(out), "n/a") {
+					fmt.Printf("Device %s doesn't support multiple queues\n", d)
+				} else {
+					channelCurrentCombined, err = strconv.Atoi(strings.TrimSpace(string(out)))
+					if err != nil {
+						testlog.Warningf(fmt.Sprintf("unable to retrieve current multi-purpose channels hardware settings for device %s on %s",
+							d, node.Name))
+					}
+					if channelCurrentCombined == 1 {
+						fmt.Printf("Device %s doesn't support multiple queues\n", d)
+					} else {
+						fmt.Printf("Device %s supports multiple queues\n", d)
+						devices[d] = channelCurrentCombined
+					}
+				}
+			}
+		}
+	}
+	// check the length of the map, if the map is empty , then
+	// there are no supported devices. Hence we skip
+	if len(devices) == 0 {
+		Skip("Skipping Test: There are no supported Network Devices")
+	}
+	return err
+}
+
+//Get Tuned Pods
+func getTunedPod(node *corev1.Node) *corev1.Pod {
```

Comment:
maybe we should extract this function and place it under the util lib since this is duplicated in https://github.com/openshift-kni/performance-addon-operators/blob/c3a829509aa75edfd7f3ef52081b6c27f96d5a3b/functests/1_performance/performance.go#L1154

Reply:
I have not yet addressed this in my latest patch . I didn't had access to cluster to test it. I will address this as soon as i have access to BM cluster.

Reply:
I tried putting in utils, but in the performance.go this function uses constants RunningOnSingleNode, testPollInterval and testPollInterval , i was not sure how to address this constants and yet keep it generic to be put in utils. Can you let me know how to address this.

Reply:
we can expand tunedForNode to get these arguments or make them available in every suit

Reply:
i have addressed this in the latest patch